src/fiatconnect/saga.ts

```diff
@@ -484,6 +484,17 @@ export function* handleSelectFiatConnectQuote({
     )
     yield put(selectFiatConnectQuoteCompleted())
     yield put(showError(ErrorMessages.PROVIDER_FETCH_FAILED))
+    const amount = {
+      crypto: new BigNumber(quote.getCryptoAmount()).toNumber(),
+      // Rounding up to avoid decimal errors from providers. Won't be
+      // necessary once we support inputting an amount in both crypto and fiat
+      fiat: Math.round(new BigNumber(quote.getFiatAmount()).toNumber()),
```

Comment:
I think this will create a bug. The FC spec says to only include fiat amount XOR crypto amount, see [here](https://github.com/fiatconnect/specification/blob/main/fiatconnect-api.md#34113-semantics). 

I suggest using crypto amount for transfers out and fiat amount for transfers in. Also, I don't think you need the `toNumber()` part; if anything that is likely to cause rounding errors.

Last, rounding up doesn't make sense, because you could exceed the user's balance. I don't think rounding is necessary if we just provide one amount, but if you still think you need to round, don't round up.

Reply:
The rounding was a sloppy copy-paste from elsewhere logic that passed without me thinking what circumstanced it applied.
There's a conversion from string involved hence the `toNumber()` and bn
I agree with the XOR part according to spec, but looks like the `handleFetchFiatConnectQuotes` saga (called by `fetchFiatConnectQuotes` action in subsequent Select Provider page) always take a crypto amount. On the other hand, `SelectProvider` requires both (in route.prarms [here](https://github.com/valora-inc/wallet/blob/b49eaa2fc66ea066b730a74e44d5bbd04a4ac8b9/src/navigator/types.tsx#L256)) so I need to figure out what to do.